### PR TITLE
Filling the `CHANGELOG.md` file

### LINKED ISSUE #59: Add missing information to the `CHANGELOG.md` file
## DoD
- Fill the `CHANGELOG.md` file with the structure according to the [keepachangelog](https://keepachangelog.com/en/1.0.0/).
- Add to the `CHANGELOG.md` all the changes made before the completion of this issue.

### PR DESCRIPTION
**Pull request by MikhailKosarev**
## What?
- Filled the `CHANGELOG.md` file with the structure according to the [keepachangelog](https://keepachangelog.com/en/1.0.0/).
- Added to the `CHANGELOG.md` all the changes made before the completion of this issue.

## Why?
To close #59

### CODE REVIEW

**Review by marinkman — COMMENTED** [review]
But we still don't keep a changelog and there are already new changes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+-  Created initial Xcode project with `MVVM` folder architecture - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
+-  Added `SwiftFormat` with rules as part of the build phase - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
+-  Added `SwiftLint` with rules as part of the build phase - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
+-  Added `.gitignore` file - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
+-  Added this `CHANGELOG.md` file - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
+-  Added an empty `README.md` file - [#30](https://github.com/ios-course/swiftcowboys-team-project/pull/30)
+-  Added images from `Figma` to the assets - [#33](https://github.com/ios-course/swiftcowboys-team-project/pull/33)
+-  Added `AppScheme` enum (with tests) according to the project color scheme - [#37](https://github.com/ios-course/swiftcowboys-team-project/pull/37)
+-  Added `Color` extension - [#37](https://github.com/ios-course/swiftcowboys-team-project/pull/37)
```
[the remark this answers is not in the document]
Comment (MikhailKosarev) [reply CHANGELOG.md]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+- Removed spaces from all project folder/file names - [#62](https://github.com/ios-course/swiftcowboys-team-project/pull/62)
+- Changed `SwiftFormat` script to support `arm64` - [#73](https://github.com/ios-course/swiftcowboys-team-project/pull/73)
+- Updated implementation of `SleepSetupView` to match the mockups - [#74](https://github.com/ios-course/swiftcowboys-team-project/pull/74)
+- Updated implementation of `DuringSleepView` to match the mockups - [#76](https://github.com/ios-course/swiftcowboys-team-project/pull/76)
+- Removed empty `SleepTrackerTests.swift` file from the `SleepTrackerTests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
+- Moved unit tests from the `SleepTrackerUITests` target into the `SleepTrackerTests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
+- Removed the `SleepTrackerUITests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
+- Added extension for the `Date` - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
```
Comment (marinkman) [inline CHANGELOG.md:41]: Why is this line in the "changed" section?

**Review by marinkman — COMMENTED** [review]
I have one question.

File: CHANGELOG.md (hunk 2 above)
Comment (MikhailKosarev) [reply CHANGELOG.md]: You're right, thanks. Moved

**Review by dersim-davaod — CHANGES_REQUESTED** [review]
Please double-check the changelog contains the correct intention of the changes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+### Changed
+
+- Filled the `README` with the description and "Authors" section - [#31](https://github.com/ios-course/swiftcowboys-team-project/pull/31)
+- Removed all images with 1x scale from project assets - [#61](https://github.com/ios-course/swiftcowboys-team-project/pull/61)
+- Removed spaces from all project folder/file names - [#62](https://github.com/ios-course/swiftcowboys-team-project/pull/62)
+- Changed `SwiftFormat` script to support `arm64` - [#73](https://github.com/ios-course/swiftcowboys-team-project/pull/73)
+- Updated implementation of `SleepSetupView` to match the mockups - [#74](https://github.com/ios-course/swiftcowboys-team-project/pull/74)
+- Updated implementation of `DuringSleepView` to match the mockups - [#76](https://github.com/ios-course/swiftcowboys-team-project/pull/76)
+- Removed empty `SleepTrackerTests.swift` file from the `SleepTrackerTests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
```
[the remark this answers is not in the document]
Comment (MikhailKosarev) [reply CHANGELOG.md:39]: @dersim-davaod why is it wrong? I can't understand, can u explain, please?

[the remark this answers is not in the document]
Comment (dersim-davaod) [reply CHANGELOG.md:39]: The description in the changelog doesn't match the changes introduced in ht PR.

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- Moved unit tests from the `SleepTrackerUITests` target into the `SleepTrackerTests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
+- Removed the `SleepTrackerUITests` target - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
+
+### Fixed
+
+- Fixed the bug after merging when initial information appeared in the`README.md` after the merging - [#32](https://github.com/ios-course/swiftcowboys-team-project/pull/32/files)
+
```
[the remark this answers is not in the document]
Comment (MikhailKosarev) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+-  Created `RoundedRectangleBlue` as `ButtonStyle` - [#70](https://github.com/ios-course/swiftcowboys-team-project/pull/70)
+-  Changed `SwiftFormat` script to support `arm64` - [#73](https://github.com/ios-course/swiftcowboys-team-project/pull/73)
+-  Updated implementation of `SleepSetupView` to match the mockups. Renamed `EstimatedSleepDurationLabelView` to `MultilineCenteredLabelView` - [#74](https://github.com/ios-course/swiftcowboys-team-project/pull/74)
+-  Updated implementation of `DuringSleepView` to match the mockups - [#76](https://github.com/ios-course/swiftcowboys-team-project/pull/76)
+-  Removed empty `SleepTrackerTests.swift` file from the `SleepTrackerTests` target. Removed the `SleepTrackerUITests` target. Unit tests from the `SleepTrackerUITests` target moved into the `SleepTrackerTests` target. Fixed unit tests (to make sure all tests pass) - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
+-  Added an extension `UserDefaults+SharedAppPreferences` for the `UserDefaults` with tests - [#84](https://github.com/ios-course/swiftcowboys-team-project/pull/84)
+
```
Comment (marinkman) [inline CHANGELOG.md:29]: Why is there an empty line here?

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+-  Added extension `Date+ShortComponents`for the `Date` - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
```
Comment (meschya) [inline CHANGELOG.md:30]:
```suggestion
-  Added extension `Date+ShortComponents`for the `Date` - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/87)
```
Incorrect reference to PR

**Review by meschya — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+-  Filled the `README` with the description and "Authors" section - [#31](https://github.com/ios-course/swiftcowboys-team-project/pull/31)
+-  Fixed the bug after merging when initial information appeared in the`README.md` - [#32](https://github.com/ios-course/swiftcowboys-team-project/pull/32/files)
+-  Added images from `Figma` to the assets - [#33](https://github.com/ios-course/swiftcowboys-team-project/pull/33)
+-  Fixed an error while building the `Sleep tracker/Tests` target - [#34](https://github.com/ios-course/swiftcowboys-team-project/pull/34)
+-  Added `Color` extension and `AppScheme` enum (with tests) according to the project color scheme - [#37](https://github.com/ios-course/swiftcowboys-team-project/pull/37)
+-  Created `SleepPageView` with subviews: `AlarmSetupView`, `EstimatedSleepDurationLabelView`, `NoAlarmLabelView`, `RoundedAlarmView`, `WheelTimePickerView`, according to the app design - [#39](https://github.com/ios-course/swiftcowboys-team-project/pull/39)
```
Comment (dersim-davaod) [inline CHANGELOG.md:14]: Did we create a `SleepPageView`?
I thought we have created `SleepSetupView` there.

File: CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+-  Created `DuringSleepView` according to the app design - [#40](https://github.com/ios-course/swiftcowboys-team-project/pull/40)
+-  Added `Image` extension and `AppImage` enum (with tests) with images according to project assets - [#51](https://github.com/ios-course/swiftcowboys-team-project/pull/51)
+-  Created `RoundedRectangleView` according to the app design - [#55](https://github.com/ios-course/swiftcowboys-team-project/pull/55)
```
Comment (dersim-davaod) [inline CHANGELOG.md:17]: nit:

```suggestion
-  Created `RoundedRectangleView` to show X/to allow Y/etc - [#55](https://github.com/ios-course/swiftcowboys-team-project/pull/55)
```

I'd suggest to have something like that.

File: CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+-  Added `RELEASENOTES.md` file with initial structure - [#58](https://github.com/ios-course/swiftcowboys-team-project/pull/58)
+-  Removed all images with 1x scale from project assets - [#61](https://github.com/ios-course/swiftcowboys-team-project/pull/61)
+-  Removed spaces from all project folder/file names - [#62](https://github.com/ios-course/swiftcowboys-team-project/pull/62)
+-  Added accidentally uncommitted changes for `SleepSetupView` - [#67](https://github.com/ios-course/swiftcowboys-team-project/pull/67)
```
Comment (dersim-davaod) [inline CHANGELOG.md:21]: nit:

```suggestion
-  Fixed the implementation for `SleepSetupView` (commited missing changes) - [#67](https://github.com/ios-course/swiftcowboys-team-project/pull/67)
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+-  Created `RoundedRectangleBlue` as `ButtonStyle` - [#70](https://github.com/ios-course/swiftcowboys-team-project/pull/70)
+-  Changed `SwiftFormat` script to support `arm64` - [#73](https://github.com/ios-course/swiftcowboys-team-project/pull/73)
```
Comment (dersim-davaod) [inline CHANGELOG.md:23]: nit:

```suggestion
-  Update `SwiftFormat` build phases to be able to run linters on `arm64` - [#73](https://github.com/ios-course/swiftcowboys-team-project/pull/73)
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+-  Updated implementation of `SleepSetupView` to match the mockups. Renamed `EstimatedSleepDurationLabelView` to `MultilineCenteredLabelView` - [#74](https://github.com/ios-course/swiftcowboys-team-project/pull/74)
+-  Updated implementation of `DuringSleepView` to match the mockups - [#76](https://github.com/ios-course/swiftcowboys-team-project/pull/76)
+-  Removed empty `SleepTrackerTests.swift` file from the `SleepTrackerTests` target. Removed the `SleepTrackerUITests` target. Unit tests from the `SleepTrackerUITests` target moved into the `SleepTrackerTests` target. Fixed unit tests (to make sure all tests pass) - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
```
Comment (dersim-davaod) [inline CHANGELOG.md:26]: This is not important:
 
```suggestion
Removed the `SleepTrackerUITests` target. Unit tests now live in the new `SleepTrackerTests` target. Fixed failed unit tests - [#82](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+-  Added an extension `UserDefaults+SharedAppPreferences` for the `UserDefaults` with tests - [#84](https://github.com/ios-course/swiftcowboys-team-project/pull/84)
```
Comment (dersim-davaod) [inline CHANGELOG.md:27]:
```suggestion
-  Added the extension `UserDefaults+SharedAppPreferences` with unit tests - [#84](https://github.com/ios-course/swiftcowboys-team-project/pull/84)
```

File: CHANGELOG.md (hunk 6 above)
Comment (MikhailKosarev) [reply CHANGELOG.md:30]: thanks, fixed

Comment (dersim-davaod) [inline CHANGELOG.md]: nit:

I know from the name the extension is for the `Date` :) 

```suggestion
-  Added the `Date+ShortComponents` extension - [#87](https://github.com/ios-course/swiftcowboys-team-project/pull/82)
```

File: CHANGELOG.md (hunk 5 above)
Comment (MikhailKosarev) [reply CHANGELOG.md:29]: thanks, didn't pay attention. Fixed

**Review by dersim-davaod — CHANGES_REQUESTED** [review]
Consider improvements:

File: CHANGELOG.md (hunk 7 above)
Comment (MikhailKosarev) [reply CHANGELOG.md:14]: you're right. Fixed

File: CHANGELOG.md (hunk 9 above)
Comment (MikhailKosarev) [reply CHANGELOG.md:21]: thanks, you're right. Changed

File: CHANGELOG.md (hunk 10 above)
Comment (MikhailKosarev) [reply CHANGELOG.md:23]: thanks, changed

File: CHANGELOG.md (hunk 11 above)
Comment (MikhailKosarev) [reply CHANGELOG.md:26]: fixed

File: CHANGELOG.md (hunk 6 above)
Comment (MikhailKosarev) [reply CHANGELOG.md]: yeah, you're right :)

File: CHANGELOG.md (hunk 12 above)
Comment (MikhailKosarev) [reply CHANGELOG.md:27]: fixed

File: CHANGELOG.md (hunk 8 above)
Comment (MikhailKosarev) [reply CHANGELOG.md:17]: I think in this lin there is no sense at all to write it. But I double-checked the rest of the lines. Thanks

Comment (dersim-davaod) [reply CHANGELOG.md]: It's up to you, but then it will be much easier to write release notes.

**Review by dersim-davaod — APPROVED** [review]
LGTM

**Review by marinkman — APPROVED** [review]
LGTM

**Review by meschya — APPROVED** [review]
LGTM

**Pull request merged**